src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -145,8 +146,11 @@ public static String getDottedPathWithAlternativeFieldNames(PersistentPropertyPa
     List<com.couchbase.client.java.query.dsl.Sort> cbSortList = new ArrayList<com.couchbase.client.java.query.dsl.Sort>();
     for (Sort.Order order : sort) {
       String orderProperty = order.getProperty();
-        //FIXME the order property should be converted to its corresponding fieldName
+      //FIXME the order property should be converted to its corresponding fieldName
       Expression orderFieldName = i(orderProperty);
+      if (order.isIgnoreCase()) {
+        orderFieldName = lower(orderFieldName);
```

Comment:
in N1QL `LOWER` on anything but a String will return null. You could use the `TOSTRING` function as well, which is accessible through `TypeFunctions.toString(Expression x)` in the DSL